tests/reals/real_slist.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef REAL_SLIST_H
+#define REAL_SLIST_H
+
+#include "macro_utils/macro_utils.h"
+#include "c_util/slist.h"
+
+#define R2(X) REGISTER_GLOBAL_MOCK_HOOK(X, real_##X);
+
+#define REGISTER_SLIST_GLOBAL_MOCK_HOOKS() \
+    MU_FOR_EACH_1(R2, \
+        slist_initialize, \
+        slist_is_empty, \
+        slist_add, \
+        slist_add_head, \
+        slist_remove, \
+        slist_remove_head, \
+        slist_find, \
+        slist_remove_if, \
+        slist_for_each \
+    )
+
+
+#include <stddef.h>
+
+
+int real_slist_initialize(PSINGLYLINKEDLIST_ENTRY list_head);
+SLIST_IS_EMPTY_RESULT real_slist_is_empty(const PSINGLYLINKEDLIST_ENTRY list_head);
```

Comment:
>SLIST [](http://example.com/codeflow?start=0&length=5)

If we're naming everything else with SINGLYLINKEDLIST, let's be consistent and name this one too.
Personally I liked SLIST more because it is shorter to type, but I am not asking you to rename, was just my remark.

Reply:
Or maybe we should rename everything to SLIST since I see the unit is named slist.
Just want to keep it consistent so we don't have to type sometimes slist sometimes singlylinkedlist

Reply:
done